src/rgw/rgw_rest_s3.cc

```diff
@@ -5128,9 +5128,14 @@ int RGWHandler_REST_S3Website::retarget(RGWOp* op, RGWOp** new_op, optional_yiel
   if (!(s->prot_flags & RGW_REST_WEBSITE))
     return 0;
 
+  if (!s->bucket) {
```

Comment:
Can you change this to:
if (rgw::sal::Bucket::empty(s->bucket)) {
instead?

Reply:
@dang I've use `s->bucket.get()` because `rgw::sal::Bucket::empty` doesn't accept the `unique_ptr` I think 🤔 
Am I corrent?

Reply:
But actually I see you added the support for `unique_ptr` as well but should it be like this?
```diff
     /** Check if a Bucket unique pointer is empty */
-    static bool empty(std::unique_ptr<Bucket> b) { return (!b || b->empty()); }
+    static bool empty(std::unique_ptr<Bucket> b) { return (!b || b.get()->empty()); }
```